atecc608a_utils.c

```diff
@@ -44,24 +44,42 @@ psa_status_t atecc608a_get_serial_number(uint8_t *buffer,
     return status;
 }
 
-psa_status_t atecc608a_check_config_locked()
+psa_status_t atecc608a_check_zone_locked(uint8_t zone)
 {
-    bool config_locked;
+    bool zone_locked;
     psa_status_t status = PSA_ERROR_GENERIC_ERROR;
 
     ASSERT_SUCCESS_PSA(atecc608a_init());
 
-    ASSERT_SUCCESS(atcab_is_locked(LOCK_ZONE_CONFIG, &config_locked));
+    ASSERT_SUCCESS(atcab_is_locked(zone, &zone_locked));
 
 exit:
     atecc608a_deinit();
     if (status == PSA_SUCCESS)
     {
-        status = config_locked? PSA_SUCCESS : PSA_ERROR_HARDWARE_FAILURE;
+        status = zone_locked? PSA_SUCCESS : PSA_ERROR_HARDWARE_FAILURE;
     }
     return status;
 }
 
+psa_status_t atecc608a_lock_data_zone()
+{
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    bool zone_locked;
+
+    ASSERT_SUCCESS_PSA(atecc608a_init());
+    ASSERT_SUCCESS(atcab_is_locked(LOCK_ZONE_DATA, &zone_locked));
```

Comment:
We don't have to do these operations inside one init-deinit block, however I decided to take advantage of it and not introduce more unnecessary code (additional initing and deiniting, error checking, etc) for a thing that's a one-liner in the end. Since this file uses driver functions directly, our function can use it, and in the application files (like `main.c`) `atecc608a_check_zone_locked` should be used instead.